### PR TITLE
Issue 19319 allow shutdown from backend

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.server.JSONP;
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:20]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.rest.api.v1.authentication.ResponseUtil'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239098)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        Logger.info(this.getClass(), "User:" + initData.getUser() + " is shutting down dotCMS!");
+        SecurityLogger.logInfo(this.getClass(),
+                        "User:" + initData.getUser() + " is shutting down dotCMS from ip:" + request.getRemoteAddr());
+
+        if (!Config.getBooleanProperty("ALLOW_DOTCMS_SHUTDOWN_FROM_CONSOLE", true)) {
+            return Response.status(Status.FORBIDDEN).build();
+        }
+
+        Runtime.getRuntime().exit(0);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:77]: Codacy found an issue: [System.exit() should not be used in J2EE/JEE apps](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239098)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * This me
+     * 
+     * @param request
+     * @param response
+     * @return
+     */
+    @DELETE
+    @Path("/shutdown")
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:56]: probably _shutdown would be better

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java (hunk 2 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:77]: is that right?

**Review by nollymar — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java]: Done

**Pull request merged**